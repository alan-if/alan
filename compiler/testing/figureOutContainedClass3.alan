The l Isa location
End The l.

Add To Every thing
  Has thing_attribute.
End Add.

Add To Every object
  Has object_attribute.
End Add.

Every c2 Isa object
  Has c2_attribute.
  Container Taking c2.
End Every.

Every c3 Isa object
  Has c3_attribute.
  Container Taking c3.
End Every.

Every c4 Isa c2
  Has c4_attribute.
  Container Taking c2.
End Every.


The cont6 Isa object
  Container Taking c4.
End The cont6.


Start At l.
  For Each i In cont6 Do
    -- i is always a c2 since cont6 might contain c4:s, and c2:s
    -- and c2 is the common parent
    If i Has thing_attribute Then "ok" End If.
    If i Has object_attribute Then "ok" End If. 
    If i Has c2_attribute Then "ok" End If.
    If i Has c3_attribute Then "error" End If.
    If i Has c4_attribute Then "error" End If.
  End For.
