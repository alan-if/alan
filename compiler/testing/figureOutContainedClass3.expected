########## figureOutContainedClass3 ##########


figureOutContainedClass3.alan

   40.      If i Has c3_attribute Then "error" End If.
=====>               1

  *1*   316 E : Attribute 'c3_attribute' is not defined for variable 'i' since
                the class it is guaranteed to be in this context ('c2') does
                not have it.

   41.      If i Has c4_attribute Then "error" End If.
=====>               1

  *1*   316 E : Attribute 'c4_attribute' is not defined for variable 'i' since
                the class it is guaranteed to be in this context ('c2') does
                not have it.


        2 error(s).
        No detected warnings.
        1 informational message(s).

